charm_versions.md

```diff
@@ -0,0 +1,138 @@
+# Versions
+
+## Juju Version
+
+The Juju version uses [semantic versioning](https://semver.org/), so is a string in the format
+major.minor.patch, where each section is a positive integer (which may be higher than 9).
+
+### Agent
+
+You can see the version of Juju used in a model in `juju status`, in the top section under
+"Version" in the tabular view, or `model.version` in JSON or YAML output. You can also use the
+`juju model-config -m controller agent-version` command.
+
+To upgrade the patch version of Juju (e.g. 3.3.0 to 3.3.1), you run
+[`juju upgrade-controller`](https://juju.is/docs/juju/juju-upgrade-controller), specifying the
+patch version. To upgrade minor or major versions, you need to create a new controller and
+[migrate](https://juju.is/docs/juju/juju-migrate) the models to it.
+
+### CLI
+
+You can see the version of the Juju CLI tool using `juju --version`. When bootstrapping a new
+controller, this is the version of the agent that will be used, unless the `--agent-version` option
+is provided.
+
+To upgrade your local Juju version, use the same tool you used to install it. For example, if you
+installed Juju with snap, you can run `snap refresh juju --channel=3.3/stable` to upgrade to
+Juju 3.3.
+
+## Charms
+
+To upgrade the version or revision of an installed charm,
+[use `juju refresh`](https://juju.is/docs/juju/manage-charms#heading--update-a-charm) for both
+local and Charmhub charms.
+
+The [jhack](https://github.com/canonical/jhack) tool has a `charm vinfo` command that will show
+the revision, channel, and workload version of a charm, as well as the versions of each charm
+lib (and if there is a more recent version available).
+
+### Charm Version
+
+Charms can optionally specify a version of the charm itself. You can find this information with
+`juju status` in the YAML or JSON formats in the `applications.<app name>.charm-version` field. If
+there is no version, the key will not be present in the status output.
+
+The version is an arbitrary string, so could be a major.minor.patch style version, a version control
+hash identifier, or anything else.
+
+To set the version, charms include a file called `version` (no extension) at the top level of the
+charm (the same folder as the `charmcraft.yaml` file), where the content of the file is the version
+string.
+
+In the past, the version would be set automatically if the charm was deployed from a known version
+control system, but that is no longer the case. The intention is that charmers set this version to
+allow tracking the installed version of the charm back to the source tree that it was built from.
```

Comment:
Our List of files in the charm project doc https://juju.is/docs/sdk/list-of-files-in-the-charm-project does not mention this. I'll investigate. Either way, the version of the charm doesn't seem to be a concept we surface in any way. E.g., on Charmhub we front channels and revisions, but there's no mention of a version. And in `juju status` (the regular, not the YAML/JSON output) Version means the workload version.